### ISSUE TITLE
Breaks with Buzz v0.17.x

### ISSUE THREAD

**Issue by NoelLH** [richsage/RMSPushNotificationsBundle#167]
There is 1 call I've noticed so far to _setTimeout()_ which is removed as per BC breaks listed at https://github.com/kriswallsmith/Buzz/blob/master/CHANGELOG.md#0170 - so as of 10 days ago a new composer install with this lib does not work.

Short term I'd suggest this lib should either pin a version range with v0.16 max, or go straight to requiring v0.17 minimum and update all these calls. I'll try to put a PR together for the latter approach today.

Sample broken call log:
> In AndroidGCMNotification.php line 83:
>                                                                               
> Attempted to call an undefined method named "setTimeout" of class "Buzz\Client\Curl".

**Comment by GenieTim** [comment]
Can confirm. What would be a temporary workaround until @richsage merges the PR?

**Comment by NoelLH** [comment]
@BernhardWebstudio You could use my branch directly but the quickest way / smallest change is probably to pin an older Buzz version in your own `composer.json`, e.g.

    ...
    "kriswallsmith/buzz": "0.16.1",
    ...

**Comment by dridi93** [comment]
Hi,

@NoelLH 

How can i solve (Attempted to call an undefined method named "setTimeout" of class "Buzz\Client\Curl".) plz.

Thanks.

**Comment by GenieTim** [comment]
Did you make the adjustment in composer.json, @dridi93 , as suggested in his [answer](url
https://github.com/richsage/RMSPushNotificationsBundle/issues/167#issuecomment-387766458) to my question? Afterwards, did you run `composer update`?

**Comment by dridi93** [comment]
@BernhardWebstudio Yes, when i did that i got "Only instances of PSR7 Response and Buzz responses are allowed"..

**Comment by dridi93** [comment]
up

**Comment by dridi93** [comment]
@NoelLH @BernhardWebstudio ... what's new !

**Comment by GenieTim** [comment]
I am sorry, with the given information, I cannot help you any further. Either you have the correct version of buzz, then there will be a method `setTimeout` and the [error](url
https://github.com/richsage/RMSPushNotificationsBundle/issues/167#issuecomment-409563262) will not occur, or you do not have the correct version of Buzz and you need to find a way to get the correct, e.g. with the [mentioned](url
https://github.com/richsage/RMSPushNotificationsBundle/issues/167#issuecomment-387766458) addition in `composer.json` before running `composer update`. If this does not work, you have to investigate; maybe you have another package requiring a different version of buzz? Maybe your error is somewhere else now?

It would be nicest, if @richsage could comment when or why not this PR could get accepted? This would eliminate this error in total...

**Comment by dridi93** [comment]
@richsage can you plz accept this PR & save my life!
